libp2p/protocols/pubsub/gossipsub/behavior.nim

```diff
@@ -199,12 +199,14 @@ proc handleIHave*(g: GossipSub,
     # TODO review deduplicate algorithm
     # * https://github.com/nim-lang/Nim/blob/5f46474555ee93306cce55342e81130c1da79a42/lib/pure/collections/sequtils.nim#L184
     # * it's probably not efficient and might give preference to the first dupe
-    var deIhaves = ihaves.deduplicate()
-    for ihave in deIhaves.mitems:
+    let deIhaves = ihaves.deduplicate()
```

Comment:
maybe we should just use a `HashSet`?

Reply:
good point but the ideal/regular size is small tho, so hashset would be just a burden, in the case of a rogue payload could be big and things would be O(n) tho